src/main/java/net/sf/jabref/model/labelpattern/GlobalLabelPattern.java

```diff
@@ -13,15 +13,31 @@
     with this program; if not, write to the Free Software Foundation, Inc.,
     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
-package net.sf.jabref.logic.labelpattern;
+package net.sf.jabref.model.labelpattern;
 
 import java.util.List;
 
+import net.sf.jabref.preferences.JabRefPreferences;
+
 public class GlobalLabelPattern extends AbstractLabelPattern {
 
+    private static List<String> defaultLabelPattern;
+
+    static {
+        updateDefaultPattern();
+    }
+
+    public static void updateDefaultPattern() {
+        defaultLabelPattern = AbstractLabelPattern.split(JabRefPreferences.getInstance().get(JabRefPreferences.DEFAULT_LABEL_PATTERN));
```

Comment:
@simonharrer Good idea
@oscargus Oh no, now I have already rewritten `GlobalLabelPattern` :-(

However, it seems you did not really touch the class in #1607? From what I have seen and tested with the GUI, you can completely drop the static part of `GlobalLabelPattern`. It is just initialized in two places (the gui and the preferences) and both of these places have access to the gui, so you can provide the correct parameters right away.

Regarding merging order: It prefer mine to be merged first :P Anyway, let Simon decide!